### PR TITLE
[6_2_X][TIMOB-17964] Android 5.0: Add support for Toolbar

### PR DESCRIPTION
**Pull request by ypbnv**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-17964

**Description:**
Backport of: https://github.com/appcelerator/titanium_mobile/pull/9147

### CODE REVIEW

**Comment by lokeshchdhry** [comment]
FR Passed.

Toolbar, its properties, methods & events from view seem to work as expected.

You need to have custom theme with no actionbar in the app:

builtin_themes.xml:
```
<?xml version="1.0" encoding="utf-8"?>
<resources xmlns:android="http://schemas.android.com/apk/res/android">
	<style name="NoActionBar" parent="Theme.AppCompat.NoTitleBar"/>
</resources>
```
tiappp.xml:
```
<android xmlns:android="http://schemas.android.com/apk/res/android">
	    <manifest>
	        <application android:theme="@style/NoActionBar"/>
	    </manifest>
</android>
```

Studio Ver: 4.9.1.201707200100
SDK Ver: 6.2.0 local build
OS Ver: 10.12.3
Xcode Ver: Xcode 8.3.3
Appc NPM: 4.2.9
Appc CLI: 6.2.3
Ti CLI Ver: 5.0.14
Alloy Ver: 1.9.13
Node Ver: 6.10.1
Java Ver: 1.8.0_101
Devices: ⇨ google Pixel --- Android 7.1.1
⇨ google Nexus 5 --- Android 6.0.1

File: apidoc/Titanium/Android/Activity.yml [hunk 1]
```diff
@@ -193,6 +193,16 @@ methods:
         type: Titanium.Android.Intent
         optional: true
 
+  - name: setSupportActionBar
+    summary: Sets a toolbar instance to be used as an ActionBar.
+    description: |
+        This method is used if you want to add a Toolbar as an ActionBar after the Activity has been created.
+        If you want to set it up before that [supportToolbar](Titanium.Android.Activity.supportToolbar) must be used.
+    parameters:
+      - name: toolbar
```
Comment (sgtcoolguy) [inline apidoc/Titanium/Android/Activity.yml]: Doc validation is failing here with missing `summary` field for the `toolbar` parameter.

File: apidoc/Titanium/UI/Android/Toolbar.yml [hunk 2]
```diff
@@ -0,0 +1,255 @@
+---
+name: Titanium.UI.Android.Toolbar
```
Comment (sgtcoolguy) [inline apidoc/Titanium/UI/Android/Toolbar.yml]: doc validation is breaking on this doc yml. I'm not entirely sure why yet...

**Comment by hansemannn** [comment]
We should really not take this PR as it is right now. It is basically creating a new - Android only - namespace that developers need to if-else-wrap with the Ti.UI.iOS.Toolbar. I made #9292 only because together with this PR, we can achieve parity to support on the shared Ti.UI.namespace. So this one should live at the same one and the docs should be added to the existing Titanium/UI/Toolbar.yml. Same thing for the open Alloy PR, this is all it should do. 

We should discuss this internally, but by merging, we won't get rid of it for a long again.

File: apidoc/Titanium/UI/Android/Toolbar.yml (hunk 2 above)
Comment (sgtcoolguy) [inline apidoc/Titanium/UI/Android/Toolbar.yml]: Finally fixed up the doc validation locally, here's what it complains about:
```
[ERROR] ./Titanium/UI/Android/Toolbar.yml
Titanium.UI.Android.Toolbar: found 7 error(s)!
	methods
		getContentInsetEndWithActions
			Invalid key(s) in Titanium.UI.Android.Toolbar: retruns
		setSubtitle
			parameters
				subtitle
					Required property "summary" not found
		setTitle
			parameters
				title
					Required property "summary" not found
		setContentInsetsAbsolute
			parameters
				insetLeft
					Required property "summary" not found
				insetRight
					Required property "summary" not found
		setContentInsetsRelative
			parameters
				insetStart
					Required property "summary" not found
				insetEnd
					Required property "summary" not found
```

**Comment by lokeshchdhry** [comment]
@hansemannn , what have we decided for this. Should this be merged ?

**Comment by hansemannn** [comment]
@lokeshchdhry Just talked to @ypbnv and we are nearly fine here, since only the docs were not updated so far. But as #9292 is merged by now, he is now able to adjust the Android docs as well, by removing the Ti.UI.Android.Toolbar docs and edit the existing Ti.UI.Toolbar docs.

**Pull request merged**